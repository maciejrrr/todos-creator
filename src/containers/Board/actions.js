import { ADD_BOARD } from './constants';

export const addBoard = ({ board }) => ({
  type: ADD_BOARD,
  board,
});
