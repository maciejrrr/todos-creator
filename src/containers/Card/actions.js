import { ADD_CARD, EDIT_CARD, ADD_TASK } from './constants';

export const addCard = ({ card }) => ({
  type: ADD_CARD,
  card,
});

export const editCard = ({ cardId, name }) => ({
  type: EDIT_CARD,
  cardId,
  name,
});

export const addTask = ({ task }) => ({
  type: ADD_TASK,
  task,
});
