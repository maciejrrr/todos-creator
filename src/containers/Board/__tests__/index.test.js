import { Board, mapDispatchToProps } from '../index';
import { addCard } from '../actions';

jest.mock('../../Card', () => 'Card');

describe('Board component', () => {
  let props;
  beforeEach(() => {
    props = {
      cards: [],
      addCard: jest.fn(),
    };
  });

  describe('snapshots', () => {
    describe('no cards', () => {
      it('renders AddCard component without any cards', () => {
        const tree = renderer.create(<Board {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('with cards', () => {
      it('renders AddCard component and cards', () => {
        const cards = [
          { id: 1, name: 'first card' },
          { id: 2, name: 'second card' },
        ];
        const tree = renderer
          .create(<Board {...props} cards={cards} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('mapDispatchToprops', () => {
    let dispatch;
    let result;

    beforeEach(() => {
      dispatch = jest.fn();
      result = mapDispatchToProps(dispatch);
    });
    describe('addCard', () => {
      it('dispatches addCard with card object', () => {
        const card = { id: 1, name: 'test' };
        result.addCard({ card });
        expect(dispatch).toBeCalledWith(addCard({ card }));
      });
    });
  });
});
