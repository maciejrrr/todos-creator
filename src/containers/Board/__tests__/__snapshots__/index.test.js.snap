// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component snapshots no cards renders AddCard component without any cards 1`] = `
<div
  className="sc-EHOje iiWQLE"
>
  <div
    className="sc-bZQynM kzxiXY"
  >
    <form
      className="sc-bxivhb gAAXBk"
      onSubmit={[Function]}
    >
      <input
        className="sc-ifAKCX cMqBtW"
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Add a card"
        value=""
      />
    </form>
  </div>
</div>
`;

exports[`Board component snapshots with cards renders AddCard component and cards 1`] = `
<div
  className="sc-EHOje iiWQLE"
>
  <div
    className="sc-bZQynM kzxiXY"
  >
    <Card
      card={
        Object {
          "boardId": 1,
          "id": 1,
          "name": "first card",
        }
      }
    />
  </div>
  <div
    className="sc-bZQynM kzxiXY"
  >
    <Card
      card={
        Object {
          "boardId": 1,
          "id": 2,
          "name": "second card",
        }
      }
    />
  </div>
  <div
    className="sc-bZQynM kzxiXY"
  >
    <form
      className="sc-bxivhb gAAXBk"
      onSubmit={[Function]}
    >
      <input
        className="sc-ifAKCX cMqBtW"
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Add a card"
        value=""
      />
    </form>
  </div>
</div>
`;

exports[`Board component snapshots with no boardId renders no board info 1`] = `
<div
  className="sc-gzVnrw dhTLwt"
>
  <h1
    className="sc-htoDjs bwZrJs"
  >
    No boards
  </h1>
</div>
`;
